### PR TITLE
Implemented Event Class

### PR DESCRIPTION
**Pull request by lealgamb**
I implemented an Event class to be used for the Activity Feed. I then added getConversation() with a given Id in ConversationStore.

Also, this is my first time making a PR from another branch so let me know if there are any problems. Just wondering if I did this correctly:

```git checkout -b NewBranch```
```git pull origin OldBranch```
```git push origin NewBranch```

### CODE REVIEW

File: src/main/java/codeu/model/data/Event.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    setUserStore(userStore.getInstance());
+    setConversationStore(conversationStore.getInstance());
+
+    if (type.equals("User")) {
+      User user = (User)classType;
+      information.add(user.getName());
+    }
+    else if (type.equals("Conversation")) {
+      Conversation conversation = (Conversation)classType;
+      information.add(userStore.getUser(conversation.getOwnerId()).getName());
```
Comment (vocatwk) [inline src/main/java/codeu/model/data/Event.java]: You can also get user by Id so you don't have to call getName()

File: src/main/java/codeu/model/data/Event.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      information.add(conversation.getTitle());
+    }
+    else if (type.equals("Message")) {
+      Message message = (Message)classType;
+      information.add(userStore.getUser(message.getAuthorId()).getName());
```
Comment (vocatwk) [inline src/main/java/codeu/model/data/Event.java]: same as above comment.

File: src/main/java/codeu/model/data/Event.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      information.add(conversationStore.getConversation(message.getConversationId()).getTitle());
```
Comment (vocatwk) [inline src/main/java/codeu/model/data/Event.java]: You can also call getConversaiton() with id since you added that method in ConversationStore

**Comment by vocatwk** [comment]
Also, about the new branch, I think it just means that you also have all the changes you had in the previous PR on this PR too. So I guess if you merge the previous first and then compare this with the master it will only have the new ones? I'm not sure if automatically updates or how that works though.

I guess you could've also pulled from master before creating this PR and then resolved conflicts after you merged your previous PR?

**Comment by yebrahim** [comment]
There are a few options for the branching issue. The simplest is: if you want to create a new PR that's dependent on an outstanding PR, you can change the base branch, see this screenshot:
image

If those changes are small and you don't mind having them as part of this PR, you can keep them here and choose to rebase when merging, see the last option in the merge dropdown:
image

File: src/main/java/codeu/model/data/Event.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    this.conversationStore = conversationStore;
+  }
+
+  /** Returns the type of this Event. */
+  public String getType() {
+    return type;
+  }
+
+  /** Returns the creation time of this Event. */
+  public Instant getCreationTime() {
```
[the remark this answers is not in the document]
Comment (lealgamb) [reply src/main/java/codeu/model/data/Event.java]: It's to display the creation time in the Activity Feed and possibly sort. The purpose of the class is to make the code cleaner and be able to group all events as one object.

File: src/main/java/codeu/model/data/Event.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  /**
+   * Constructs a new Event
+   * 
+   * @param type the type of event
+   * @param creation the time the event was created
+   * @param classType the class of the event 
+   */
+  public Event(String type, Instant creation, Object classType) {
+    this.type = type;
+    this.creation = creation;
```
[the remark this answers is not in the document]
Comment (lealgamb) [reply src/main/java/codeu/model/data/Event.java]: It's to have access to the creation time of the actual event, like a user joining, not when the Event object is created.

**Comment by lealgamb** [comment]
Thanks for the help with the branching. Just to make sure, what I did didn't create any problems, right?

**Comment by yebrahim** [comment]
All looks good.
It's a good practice to create a branch per PR. Reusing branches can be tricky and you can accidentally shoot yourself in the foot.

**Pull request merged**